raft/raft_log.cpp

```diff
@@ -0,0 +1,165 @@
+// Copyright (C) 2018 Bluzelle
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License, version 3,
+// as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+#include "raft_log.hpp"
+
+#include <fstream>
+#include <boost/filesystem/operations.hpp>
+#include <iostream>
+
+
+namespace bzn
+{
+    raft_log::raft_log(const std::string& log_path) : entries_log_path(log_path)
+    {
+        if(boost::filesystem::exists(this->entries_log_path))
+        {
+            std::ifstream is(this->entries_log_path, std::ios::in | std::ios::binary);
+            bzn::log_entry log_entry;
+            while (is >> log_entry)
+            {
+                this->log_entries.emplace_back(log_entry);
+            }
+            is.close();
```

Comment:
destructor will close

Reply:
Done. Removed on of suspenders or belt...